# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/mips32/Lowering-mips32.h
# Commit: eb35fdc50799
# Full Hash: eb35fdc507996da61eaceffeb9f9e9b12f3ba0ca
# Author: Dmitry Bezhetskov <dbezhetskov@igalia.com>
# Date: 2020-09-10 15:21:10
# Regressor Bug: 1639153
# File Overlap Count: 1
# Description:
#   Bug 1639153 - Part 6.2: Establish dependency from tls for x86 callWithABI div/mod i64. r=lth
#   
#   x86 has few register so to do div/mod for i64 it call the runtime
#   and clobber almost all gp registers including WasmTlsReg.
#   To be able to call c++ runtime via Builtin thunk we need to set up WasmTlsReg.
# ==============================================================================

diff -r 3952f1301e38 -r eb35fdc50799 js/src/jit/mips32/Lowering-mips32.h
--- a/js/src/jit/mips32/Lowering-mips32.h	Thu Sep 10 02:41:27 2020 +0000
+++ b/js/src/jit/mips32/Lowering-mips32.h	Thu Sep 10 05:18:04 2020 +0000
@@ -34,7 +34,9 @@
   void lowerTruncateFToInt32(MTruncateToInt32* ins);
 
   void lowerDivI64(MDiv* div);
+  void lowerWasmBuiltinDivI64(MWasmBuiltinDivI64* div);
   void lowerModI64(MMod* mod);
+  void lowerWasmBuiltinModI64(MWasmBuiltinModI64* mod);
   void lowerUDivI64(MDiv* div);
   void lowerUModI64(MMod* mod);
 };